modules/generator/generator.go

```diff
@@ -369,6 +383,10 @@ func (g *Generator) createInstance(id string) (*instance, error) {
 }
 
 func (g *Generator) CheckReady(_ context.Context) error {
+	if !g.cfg.JoinRing {
```

Comment:
there's a lot of small behavior changes scattered around due to this "JoinRing" boolean. obviously this is related to one of my previous comments. I'm fine with things being messy in the interim before we land on a final pattern that both adaptive traces and rhythm can use, but we really need comments for lines like this to help us understand why we would behave this way when JoinRing = false

Reply:
I have tried to find better names for the config flags and introduced a helper `shouldJoinRing` to hopefully make this kind of behaviour change this less confusing.